### PR TITLE
Update nan dependency to 2.0.x

### PR DESCRIPTION
**Pull request by bnoordhuis**
This is needed for compatibility with the latest io.js release(s).

@3rd-Eden Can you do a release for this?  Thanks.

### CODE REVIEW

File: src/hashvalue.cc [hunk 1]
```diff
@@ -3,20 +3,18 @@
 using namespace v8;
 
 NAN_METHOD(Hasher) {
-  NanScope();
+  unsigned int hash = ((int) info[0]->NumberValue() << 24)
+    | ((int) info[1]->NumberValue() << 16)
+    | ((int) info[2]->NumberValue() << 8)
+    | (int) info[3]->NumberValue();
```
Comment (bnoordhuis) [inline src/hashvalue.cc]: On a slightly related subject, is there a reason you don't use Int32Value() or Uint32Value() here?

I think you can even do this entirely in JS land, V8 should be able to lower it to pretty much the same code a C++ compiler would generate, with less overhead even.

**Comment by bnoordhuis** [comment]
Hrm, this is mildly annoying.  io.js v3.x needs gcc 4.8 for its C++11 support but adding that to the .travis.yml somehow makes istanbul disappear from the PATH...

**Comment by bnoordhuis** [comment]
Reimplemented the hash in pure JS and the CI is all green now.  Seems to be slightly faster too, at least on my system.

**Comment by 3rd-Eden** [comment]
@bnoordhuis Thanks for the pull request, but I have no idea why tests are passing as I'm pretty sure that hashing required 64 bit bit shifting in order to correctly generate a stable hashring. And that was the reason I moved to a binary addon in the first place.

**Comment by bnoordhuis** [comment]
The C++ code uses (unsigned) int and that's 32 bits on all platforms node runs on.

**Comment by 3rd-Eden** [comment]
That is an excelent point lol.

**Pull request merged**